packages/inertia-react/src/useForm.js

```diff
@@ -3,14 +3,15 @@ import useRemember from './useRemember'
 import { Inertia } from '@inertiajs/inertia'
 import { useCallback, useEffect, useRef, useState } from 'react'
 
-export default function useForm(...args) {
+export default function useForm(initialData, rememberKey) {
```

Comment:
Can we flip the arguments of this back to how it was? Essentially, `rememberKey` first, and the `initialData` second?

The main reason we've decided to do the `rememberKey` first, has to do with how it looks/reads when you define your form:

```js
form: this.$inertia.form({
  provider: '',
  name: '',
  size: '2048',
  region: null,
  seller: 'online',
  recipe: '',
  is_public: false,
}, 'remember-key'),

```

vs:

```js
form: this.$inertia.form('remember-key', {
  provider: '',
  name: '',
  size: '2048',
  region: null,
  seller: 'online',
  recipe: '',
  is_public: false,
}),
```

On larger forms, the 2nd is a lot more identifiable/readable by the first line.

Apart from this 'preference', we also want to keep all adapters on-par in terms of the features they offer, and how the feature works, so that anyone that works in multiple languages can rely on their muscle memory as much as possible when working with Inertia.

Thanks!

Reply:
Hi Claudio!

This does generate a few issues. The first param can be either a rememberKey or the form you want to remember. This can be solved in the code, but makes the typing confusing.

I updated the pull request with your comments processed, but maybe this is something that should be looked at in future releases.